engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala

```diff
@@ -110,20 +110,29 @@ case class WorkflowExecutionActor(workflowDescriptor: EngineWorkflowDescriptor,
     case Event(BypassedDeclaration(declKey), stateData) =>
       handleDeclarationEvaluationSuccessful(declKey, WdlOptionalValue.none(declKey.scope.wdlType), stateData)
 
-      // Failure
-        // Initialization
+    // Failure
+    // Initialization
     case Event(JobInitializationFailed(jobKey, reason), stateData) =>
       pushFailedCallMetadata(jobKey, None, reason, retryableFailure = false)
       handleNonRetryableFailure(stateData, jobKey, reason, Map.empty)
-      // Job Non Retryable
+    // Job Non Retryable
     case Event(JobFailedNonRetryableResponse(jobKey, reason, returnCode), stateData) =>
       pushFailedCallMetadata(jobKey, returnCode, reason, retryableFailure = false)
       handleNonRetryableFailure(stateData, jobKey, reason, Map.empty)
-      // Job Retryable
+    // Aborted? But we're outside of the AbortingState!?? Could happen if
+      // - The job was aborted by something external to Cromwell
+      // - The job lasted too long (eg JES 6 day timeout)
+      // - We've reconnected to an aborting job (some sort of shutdown race condition?)
+    // Treat it like any other non-retryable failure:
+    case Event(AbortedResponse(jobKey), stateData) =>
+      val cause = new Exception("The job was aborted from outside Cromwell")
```

Comment:
I know you've already merged this but this could use a more explanatory message IMO. Now we'll see if I care enough to bother making a PR. I think I know the answer.